rust/private/providers.bzl

```diff
@@ -27,6 +27,10 @@ CrateInfo = provider(
         "root": "File: The source File entrypoint to this crate, eg. lib.rs",
         "rustc_env": "Dict[String, String]: Additional `\"key\": \"value\"` environment variables to set for rustc.",
         "srcs": "depset[File]: All source Files that are part of the crate.",
+        "transitive_type": (
+            "str, optional: The original crate type for targets generated using a previously defined " +
```

Comment:
It seems this is the only optional field in this provider, and you seem to be adding it everywhere explicitly. Are you trying to make the code backwards compatible?

Reply:
I've never introduced an optional parameter on a provider before so I had assumed I needed to plug it in everywhere. But Assuming I don't set it, does it default to `None`?

Reply:
By leaving this blank, I run into
```
Error: 'CrateInfo' value has no field or method 'wrapped_crate_type'
Available attributes: aliases, deps, edition, is_test, name, output, proc_macro_deps, root, rustc_env, srcs, type
```

It seems the field needs to be explicitly created.

Reply:
That is very unfortunate. This makes this a backwards incompatible change (and the field is not optional so please fix the documentation).

This could have been prevented if we had `rust_common.create_crate_info` function. I propose following:

1. let's create `rust_common.create_crate_info` function. I'm happy to do it if you want.
2. let's update the docs of `rust_common.crate_info` and say this is not supposed to be used for construction, only for accessing the provider (unfortunately provider constructor is used as both constructor and type declaration that we use in rule definitions or when getting providers from target).
2. let's call this function in all the places in this repo where we currently call `rust_common.crate_info(...)`.
3. let's recommend people migrating for this PR to use `create_crate_info` instead of `crate_info`.

Reply:
Was https://github.com/bazelbuild/rules_rust/pull/818 what you were thinking here?